pkg/libcoap/0006-fix-option-parsing-with-debug-enabled.patch

```diff
@@ -0,0 +1,29 @@
+From c64b0a837ed752fa76c9bef2226c7a05fd5b1e6f Mon Sep 17 00:00:00 2001
+From: Oleg Hahm <oleg@hobbykeller.org>
+Date: Thu, 23 Jul 2015 10:22:26 +0200
+Subject: [PATCH 6/7] fix option parsing with debug enabled
+
+---
+ option.c | 5 +++--
+ 1 file changed, 3 insertions(+), 2 deletions(-)
+
+diff --git a/option.c b/option.c
+index a03fc9e..b2864c4 100644
+--- a/option.c
++++ b/option.c
+@@ -57,9 +57,10 @@ coap_opt_parse(const coap_opt_t *opt, size_t length, coap_option_t *result) {
+ 
+   switch(result->delta) {
+   case 15:
+-    if (*opt != COAP_PAYLOAD_START)
++    if (*opt != COAP_PAYLOAD_START) {
+       debug("ignored reserved option delta 15\n");
+-    return 0;
++      return 0;
```

Comment:
Well, actually the question is if the curly brackets are missing on purpose or not. I'll open an issue at libcoap repository.